lib/Sema/CSSimplify.cpp

```diff
@@ -1348,8 +1348,9 @@ ConstraintSystem::matchTupleTypes(TupleType *tuple1, TupleType *tuple2,
   return getTypeMatchSuccess();
 }
 
-// Returns 'false' (i.e. no error) if it is legal to match functions with the
-// corresponding function type representations and the given match kind.
+// Returns 'true' (i.e. representations match) if it is legal to match
+// functions with the corresponding function type representations and
+// the given match kind.
 static bool matchFunctionRepresentations(FunctionTypeRepresentation rep1,
```

Comment:
Let's preserve the true-on-error return. It's a bit backwards, but funnily enough is *more* inconsistent to make this change.

Reply:
Thanks @CodaFi, I'm on my way to revert it :)